tExtRef/extRefTypeRestrictions.ts

```diff
@@ -1,4 +1,16 @@
-import dataObjects from "../foundation/nsd.json";
+async function fetchJSON(path: string): Promise<unknown> {
+  const response = await fetch(new URL(path, import.meta.url));
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch JSON file: ${path.toString()} ${response.status} ${
+        response.statusText
+      }`
+    );
+  }
+  return await response.json();
+}
+
+const dataObjects = await fetchJSON("../foundation/nsd.json");
```

Comment:
What's the functional difference between this and

```suggestion
const dataObjects = await fetch(new URL("../foundation/nsd.json", import.meta.url)).then(res => res.json());
```

apart from the custom error message? I'm asking quite naively, I assume that both would just result in a rejected promise in case of either a non-ok response or invalid json, but I haven't tried. If and only if there is a significant functional advantage I would keep the custom function, otherwise I'd probably prefer the one-liner for readability.

Reply:
I agree, let's do that.